Sumo-Logic-Tools/Event_Based_S3_Automation/script.py

```diff
@@ -0,0 +1,265 @@
+import requests
+import boto3
+import json
+from requests.auth import HTTPBasicAuth
+from collections import defaultdict
+
+'''
+Helps in taking in the inputs required for the script to function properly.
+There are 3 things the user has to enter.
+1. Select the API endpoint. https://help.sumologic.com/APIs/General-API-Information/Sumo-Logic-Endpoints-and-Firewall-Security can help in making the selection.
+2. Enter the Sumo accessId.
+3. Enter the Sumo accessKey.
+https://help.sumologic.com/Manage/Security/Access-Keys can help with the second and third step.
+'''
+class InitialSetupHelper:
+
+	apiEndpointOrderedList = ['https://api.us2.sumologic.com/api', 'https://api.sumologic.com/api', 'https://api.au.sumologic.com/api', 'https://api.eu.sumologic.com/api', 'https://api.de.sumologic.com/api']
+
+	def takeRequiredInputs(self):
+		print("\n WELCOME! THIS SCRIPT WILL AUTOMATE THE EVENT BASED S3 SETUP FOR YOU. IT WILL ADD THE NOTIFICATION BASED MECHANISM ON ALL OF YOUR POLLING SOURCES! \n")
+		print("\nBUT FIRST YOU NEED TO PROVIDE SOME INPUTS \n")
+		print("\nMAKE SURE THAT YOU HAVE RUN AWS CONFIG AND CONFIGURED THE AWS CLI PROPERLY WITH THE CORRECT CREDENTIALS AND REGION \n")
+		print("\nWHICH SUMO API ENDPOINT DO YOU GUYS USE? [1-5]")
+		print("1. ", self.apiEndpointOrderedList[0], "\n2. ", self.apiEndpointOrderedList[1], "\n3. ", self.apiEndpointOrderedList[2], "\n4. ", self.apiEndpointOrderedList[3], "\n5. ", self.apiEndpointOrderedList[4])
+
+		while True:
+			deployment = int(input("Enter your choice: "))
+			if (deployment < 1 or deployment > 5):
+				print("Please select a value between 1 to 5")
+				continue
+			else:
+				break
+
+		base_url = str(self.apiEndpointOrderedList[deployment - 1])
+
+		print("\nEnter you SUMO ACCESS ID AND ACCESS KEY")
+
+		access_id = input("ACCESS ID: ")
+		access_key = input("ACCESS KEY: ")
```

Comment:
We should use a masked input API to read access key, it should not show in the terminal as plaintext